crypto/BRCryptoWalletManager.c

```diff
@@ -945,9 +937,33 @@ cryptoWalletManagerSubmitSigned (BRCryptoWalletManager cwm,
         }
 
         case BLOCK_CHAIN_TYPE_GEN: {
+            // We don't have GEN Events bubbling up that we can handle by adding transfer
+            // to wallet.  So, we'll add transfer here...
+            cryptoWalletAddTransfer (wallet, transfer);
```

Comment:
Should the actual submit happen first - and perhaps get some result back from the server before adding to the wallet?